rivescript/parser.py

```diff
@@ -574,6 +574,14 @@ def check_syntax(self, cmd, line):
                     angle += 1
                 elif char == '>':
                     angle -= 1
+                elif char == '|':
+                    if parens == 0 and square == 0:   # Pipe outside the alternative and option
+                        return "Pipe | must be within parenthesis brackets or square brackets"
+
+                total = parens + square + curly + angle   # At each character, not more than 1 bracket opens
+                for special_char_count in [parens, square, curly, angle, total]:
+                    if special_char_count not in (0, 1):
+                        return "Unbalanced brackets"
```

Comment:
Would this raise syntax errors on a trigger like `+ my name is (<bot name>|<bot master>)`? It seems to check that there is _at most_ one active bracket open, per character scanned (the checks a couple lines further down check the counts _after_ the whole trigger has been scanned, to check if there are any dangling mismatched brackets at the end).

Another example trigger that'd be plausibly in a bot would be an angled tag inside an optional, like idk,

```
+ [<bot name>] *
- You called? <@>
```

I think the better fix for #86 is to move the syntax check call to after the `^Continue` is handled, around [this part of the code](https://github.com/aichaos/rivescript-python/blob/858dd6df4f92987323ee6c6f2fa3d69964d6b93a/rivescript/parser.py#L195-L218). Here, we're calling `check_syntax` near the top and then looking ahead after that. (This would mean the bug also extends to _any_ command with strict syntax rules, like if you continued a `! Definition` or `*Condition` in the wrong spots)

For fixing the syntax checker itself, it would be ideal if it could detect things like "an opening ( with a closing ] somewhere inside it", like `+ [ some ( invalid ] ordering )`, which raise a regexp syntax error but my code doesn't catch it because it's all balanced. But I don't know how the code could check that while not looking like a tangled mess.